src/node/internal/internal_timers.ts

```diff
@@ -0,0 +1,252 @@
+      this.#timer = globalThis.setInterval(
+        this.#callback,
+        this.#after,
+        ...this.#args
+      );
+    } else {
+      // @ts-expect-error TS2322 Due to difference between Node.js and globals
+      this.#timer = globalThis.setTimeout(
+        this.#callback,
+        this.#after,
+        ...this.#args
+      );
+    }
+  }
+
+  #clearTimeout(): void {
+    if (this.#isRepeat) {
+      globalThis.clearInterval(this.#timer);
+    } else {
+      globalThis.clearTimeout(this.#timer);
+    }
+  }
+
+  public refresh(): this {
+    this.#clearTimeout();
+    this.#constructTimer();
+    return this;
+  }
+
+  public unref(): this {
+    // Intentionally left as no-op.
+    this.#isRefed = false;
+    return this;
+  }
+
+  public ref(): this {
+    // Intentionally left as no-op.
+    this.#isRefed = true;
+    return this;
+  }
+
+  public hasRef(): boolean {
+    return this.#isRefed;
+  }
+
+  public close(): this {
+    this.#clearTimeout();
+    return this;
+  }
+
+  public [Symbol.dispose](): void {
+    this.#clearTimeout();
+  }
+
+  public [Symbol.toPrimitive](): number {
+    return this.#timer;
+  }
+
+  static {
+    clearTimeoutImpl = (obj: Timeout): void => {
+      obj.#clearTimeout();
+    };
+  }
+}
+
+export class Immediate {
+  // @ts-expect-error TS2724 Node.js and global difference.
+  #timer: globalThis.Immediate;
+  #hasRef: boolean = false;
+
+  public constructor(
+    callback: (...args: unknown[]) => void,
+    args: unknown[] = []
+  ) {
+    this.#timer = globalThis.setImmediate(callback, ...args);
+  }
+
+  public ref(): this {
+    // Intentionally left as no-op.
+    this.#hasRef = true;
+    return this;
+  }
+
+  public unref(): this {
+    // Intentionally left as no-op.
+    this.#hasRef = false;
+    return this;
+  }
+
+  public hasRef(): boolean {
+    return this.#hasRef;
+  }
+
+  public [Symbol.dispose](): void {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+    globalThis.clearImmediate(this.#timer);
+  }
+
+  static {
+    clearImmediateImpl = (obj: Immediate): void => {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+      globalThis.clearImmediate(obj.#timer);
+    };
+  }
+}
+
+export function setTimeout(
+  callback: (...args: unknown[]) => unknown,
+  after: number,
+  ...args: unknown[]
+): Timeout {
+  validateFunction(callback, 'callback');
+
+  return new Timeout(
+    callback,
+    after,
+    args,
+    /* isRepeat */ false,
+    /* isRefed */ true
+  );
+}
+
+export function clearTimeout(timer: unknown): void {
+  if (timer instanceof Timeout) {
+    clearTimeoutImpl(timer);
+  } else if (typeof timer === 'number') {
+    globalThis.clearTimeout(timer);
+  }
+}
+
+export function setImmediate(
+  callback: (...args: unknown[]) => void,
+  ...args: unknown[]
+): Immediate {
+  validateFunction(callback, 'callback');
+  return new Immediate(callback, args);
+}
```

Comment:
This should just be `export const setImmediate = globalThis.setImmediate;`

Reply:
globalThis.setImmediate returns an ID, whereas we want to return an Immediate class in here. This wouldn't be compatible with nodejs, if I'm not missing anything here?